pkg/client/restclient/request.go

```diff
@@ -786,22 +804,31 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 	// TODO: Change to a timeout based approach.
 	maxRetries := 10
 	retries := 0
+	var req *http.Request
+	var resp *http.Response
 	for {
-		url := r.URL().String()
-		req, err := http.NewRequest(r.verb, url, r.body)
-		if err != nil {
-			return err
-		}
-		req.Header = r.headers
+		// TODO dedup multiple usage of WrapWithRetry in Stream/Watch/request methods
+		err := r.urlProvider.WrapWithRetry(
```

Comment:
Looking to this endless log of comments for endless review, just wondering if accepting the *basic* implementation for multi-hosts access and *then* improving it with rate limiting and other things would make sense? The thing is very nice to have. F.e. etcd cluster can already be accessed in multi-hosts fashoin by *all* cluster components including network plugins. This allowed us (in Kargo, https://github.com/kubernetes-incubator/kargo/pull/591) to drop etcd-proxy from HA layout and just live with multi-access mode. The same is expected by ops with Apiserver.